rust/src/lib.rs

```diff
@@ -790,6 +792,62 @@ pub extern "C" fn zcashlc_get_address(
     unwrap_exc_or_null(res)
 }
 
+struct UnifiedAddressParser(UnifiedAddress);
+
+impl zcash_address::TryFromRawAddress for UnifiedAddressParser {
```

Comment:
This exists because we don't have this implemented on `UnifiedAddress` itself (only on `ReceiverAddress`), and it's faster to just do this to unblock the PR. We can move this into `zcash_client_backend` at some point.

Reply:
An alternative is to use the parsing code `PaymentAddress` and then just pattern match on the result. That would be a tiny bit more flexible, as it'd work both with taddr strings and UA strings.

Reply:
I assume you mean the `ReceiverAddress` parsing code (as `PaymentAddress` is specifically Sapling). But we don't want this to work for taddr strings? For those it would be a no-op, and a success there would mislead the caller, who is expecting to receive a component of a Unified Address.

Reply:
Ah, actually I meant `RecipientAddress`; `ReceiverAddress` isn't a thing and I was confused. Anyway, one could still return an error for taddr strings if you wanted to, and it'd avoid the `zcash-address` dependency.